t/plugin/authz-casbin.t

```diff
@@ -0,0 +1,358 @@
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+use t::APISIX 'no_plan';
+
+repeat_each(1);
+no_long_string();
+no_root_location();
+run_tests;
+
+__DATA__
+
+=== TEST 1: sanity
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.authz-casbin")
+            local conf = {
+                model_path = "/path/to/model.conf",
+                policy_path = "/path/to/policy.csv",
+                username = "user"
+            }
+            local ok, err = plugin.check_schema(conf)
+            if not ok then
+                ngx.say(err)
+            end
+            ngx.say("done")
+        }
+    }
+--- request
+GET /t
+--- response_body
+done
+--- no_error_log
+[error]
+
+
+
+=== TEST 2: username missing
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.authz-casbin")
+            local conf = {
+                model_path = "/path/to/model.conf",
+                policy_path = "/path/to/policy.csv"
+            }
+            local ok, err = plugin.check_schema(conf)
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("done")
+            end
+        }
+    }
+--- request
+GET /t
+--- response_body
+property "username" is required
+--- no_error_log
+[error]
+
+
+
+=== TEST 3: put model and policy text in metadata
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.authz-casbin")
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/plugin_metadata/authz-casbin',
+                ngx.HTTP_PUT,
+                [[{
+                    "model": "[request_definition]
+                    r = sub, obj, act
+
+                    [policy_definition]
+                    p = sub, obj, act
+
+                    [role_definition]
+                    g = _, _
+
+                    [policy_effect]
+                    e = some(where (p.eft == allow))
+
+                    [matchers]
+                    m = (g(r.sub, p.sub) || keyMatch(r.sub, p.sub)) && keyMatch(r.obj, p.obj) && keyMatch(r.act, p.act)",
+
+                    "policy": "p, *, /, GET
+                    p, admin, *, *
+                    g, alice, admin"
+                }]]
+                )
+
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 4: Enforcer from text without files
```

Comment:
Test 4 already contains Test 3?

Reply:
Yes, removed that unnecessary part from test 4.